apps/web/pages/settings/my-account/out-of-office/index.tsx

```diff
@@ -375,20 +375,30 @@ const OutOfOfficePage = () => {
   }, [openModalOnStart]);
 
   const [openModal, setOpenModal] = useState(false);
+
+  const { data } = trpc.viewer.outOfOfficeReasonList.useQuery();
+  // TODO: The best solution is we should find the cause of full page flicker
+  // Taling about hydration problem, migrating to appDir might solve the problem automatically
+  // which will reset the Modal state, which makes bad UX and fails (superhuman click speed) E2E tests
+  const isBeforeFlicker = data?.length === undefined || data.length <= 0;
```

Comment:
A hydration error from Next.js popped up before this fix sometimes, preventing the playwright from clicking anything and waiting infinitely. I suspect this because it stated it caused a flicker, but further investigation is suggested.
https://github.com/calcom/cal.com/blob/5b7f2a6ffcfcda3aa9411fe482fd58519a76ed07/packages/ui/components/meta/Meta.tsx#L52